risc0/zkvm/src/host/server/prove/mod.rs

```diff
@@ -52,7 +53,40 @@ pub trait ProverServer {
     ) -> Result<ProveInfo> {
         let mut exec = ExecutorImpl::from_elf(env, elf)?;
         let session = exec.run()?;
-        self.prove_session(ctx, &session)
+        let prove_info = self.prove_session(ctx, &session)?;
```

Comment:
I would expect that if I call `prover.prove_session` that it would also follow the `ReceiptFormat` (aka `ReceiptKind`). So it seems like `prove_session` should take the `ProverOpts` as a parameter.